### PR TITLE
Fixes certificate expiry issue, renew the expired certificates

### PR DESCRIPTION
**Pull request by gunhansancar**
Fixes certificate expiry issue, renew the expired certificates, and adds test cases

### CODE REVIEW

File: common/services/deviceOperation.js [hunk 1]
```diff
@@ -48,22 +49,71 @@ const _signer = function (deviceId, operation) {
...
+ *
+ * @param {string} deviceId - The ID of the device
+ * @param {string} operation - The operation to execute
+ * @returns {Promise} Axios response from the device execution request
+ * @throws {Object} Verification error if 2FA is required
+ */
+const _executor = async function (deviceId, operation) {
+    try {
+        // Validate or register the certificate before proceeding
+        await validateAndRegisterCertificate();
```
Comment (RafaRuiz) [inline common/services/deviceOperation.js:96]: @mpbarnwell this is the only line added to the executor, are you ok with this?

File: package.json [hunk 2]
```diff
@@ -1,6 +1,7 @@
 {
   "name": "@doordeck/javascript-doordeck-sdk",
   "title": "Javascript Doordeck Sdk",
+  "type": "module",
```
Comment (RafaRuiz) [inline package.json:4]: needed?

Comment (RafaRuiz) [reply package.json:4]: ah I see the `node --experimental-vm-modules`

**Review by RafaRuiz — APPROVED** [review]

File: package.json (hunk 2 above)
Comment (gunhansancar) [reply package.json:4]: This was needed to be able to run the tests. But, if it makes an issue, we can keep some changes local

File: common/services/deviceOperation.js (hunk 1 above)
Comment (mpbarnwell) [reply common/services/deviceOperation.js:96]: I don't know JavaScript well enough to answer that, what are the implications?

Comment (RafaRuiz) [reply common/services/deviceOperation.js:96]: it makes this check on every execution
image
which isn't a bad choice but we don't do it in the mobile sdks

Comment (mpbarnwell) [reply common/services/deviceOperation.js:96]: Ok, does retrieveSavedCert check the expiry as well?

Comment (gunhansancar) [reply common/services/deviceOperation.js:96]: Yes, `retrieveSavedCert` actually does the `endDate` check as well. 
It returns null if anything goes wrong.
So, if the certificate is there and it is valid `if(validCert) return true;` is an early return from this validate function.

**Review by RafaRuiz — APPROVED** [review]

**Pull request merged**